synapse/handlers/directory.py

```diff
@@ -360,22 +374,58 @@ def can_modify_alias(self, alias, user_id=None):
         return defer.succeed(True)
 
     @defer.inlineCallbacks
-    def _user_can_delete_alias(self, alias, user_id):
+    def _user_can_delete_alias(self, alias: RoomAlias, user_id: str):
+        """Determine with a user can delete an alias.
+
+        One of the following must be true:
+
+        1. The user created the alias.
+        2. The user is a server administrator.
+        3. The user has a power-level sufficient to send a canonical alias event
+           for the current room.
+
+        """
         creator = yield self.store.get_room_alias_creator(alias.to_string())
 
         if creator is not None and creator == user_id:
             return True
 
         is_admin = yield self.auth.is_server_admin(UserID.from_string(user_id))
-        return is_admin
+        if is_admin:
```

Comment:
can we not just use `self.auth.check_can_change_room_list` here?

Reply:
Looks like we can. That would change behavior slightly:
* Currently this code does not require the user to be *in* the room, only to have the proper power level. This might be more correct anyway?
* This method would then potentially raise instead of always returning a boolean. The user facing change of this is that the error message would sometimes be "This server requires you to be a moderator in the room to  edit its room list entry" instead of "Unknown room alias". This is probably fine, just thought I'd mention it.

Reply:
Note that it doesn't seem necessary to be in a room to add an alias.

Reply:
Discussed this on Matrix a bit. Requiring the user to be in the room should be OK.

I misspoke on the second point, the error message would be "You don't have permission to delete the alias.", but my question is still valid. We decided returning a boolean from `auth.check_can_change_room_list` likely makes sense so that each consumer can raise their own errors.